### ISSUE TITLE
Between version 1.7.22 and 1.7.23 query to pring Tag fails

### ISSUE THREAD

**Issue by gergnz** [aws/aws-cli#1306]
This command works and prints out the value of the tag in 1.7.22
```bash
aws --output text --region ap-southeast-2 ec2 describe-instances --instance-ids i-aed46472 --query 'Reservations[*].Instances[*].Tags[?Key==`Environment`].Value[]'
```

In 1.7.23 and 1.7.24 it prints nothing.

**Comment by jamesls** [comment]
I'm seeing the same behavior, this is a regression.  The workaround until a fix is out is to use `[]` instead of `[*]`:
```
'Reservations[].Instances[].Tags[?Key==`Environment`].Value[]'
```

**Comment by jamesls** [comment]
Issue has been fixed and will be available in the next CLI release.  Thanks for reporting.

**Comment by jamesls** [comment]
Fix is now available in v1.7.25 of the AWS CLI.
